### PR TITLE
doc: add initial CODEOWNERS and MAINTAINERS

### PR DESCRIPTION
**Pull request by FeynmanZhou**
Add initial CODEOWNERS and MAINTAINERS per discussion in https://github.com/notaryproject/.github/issues/26. I hope @notaryproject/notaryproject-governance-maintainers @notaryproject/notaryproject-org-maintainers could help to review and approve this PR. Thanks!

### CODE REVIEW

File: MAINTAINERS [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Repo-Level Maintainers (in alphabetical order)
+# Note: This is for the notaryproject/notation-hashicorp-vault repo
+Alexander Scheel <alex.scheel@hashicorp.com> (@cipherboy)
+Bingqi Shang <bingqi.shang.apply@gmail.com> (@OliverShang)
+Patrick Zheng <patrickzheng@microsoft.com> (@patrickzheng200)
```
Comment (toddysm) [inline MAINTAINERS]: This user doesn't match the CODEOWNERS entry. Also the new handle is @Two-Hearts

**Review by toddysm — CHANGES_REQUESTED** [review]

File: CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Repo-Level Owners (in alphabetical order)
+# Note: This is only for the notaryproject/notation-hashicorp-vault repo
+* @cipherboy @OliverShang @priteshbandi @shizhMSFT
```
Comment (ghost) [inline CODEOWNERS]: Hi @FeynmanZhou, should I be added into this list?

**Review by ghost — APPROVED** [review]
LGTM

**Review by toddysm — APPROVED** [review]

**Review by cipherboy — APPROVED** [review]
Not sure if I'm supposed to ACK this or not, but my details look good and as I've mentioned to Feynman in a mail, happy to help review designs and code to the project!

**Review by priteshbandi — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**